simple/stats/config.py

```diff
@@ -120,7 +129,32 @@ def database(self) -> dict:
     return self.data.get(_DATABASE_FIELD)
 
   def _input_file(self, input_file_name: str) -> dict:
-    return self.data.get(_INPUT_FILES_FIELD, {}).get(input_file_name, {})
+    # Exact match.
+    input_file_config = self._input_files_config.get(input_file_name, {})
+    if input_file_config:
+      return input_file_config
+
+    # Wildcard match
+    if input_file_name not in self._input_file_name_keys.keys():
+      self._input_file_name_keys[input_file_name] = self._input_file_name_match(
+          input_file_name)
+    return self._input_files_config.get(
+        self._input_file_name_keys[input_file_name], {})
+
+  def _input_file_name_match(self, input_file_name: str) -> str | None:
+    for input_file_pattern in self._input_files_config.keys():
+      if "*" not in input_file_pattern:
+        continue
+      regex = self._input_file_pattern_to_regex(input_file_pattern)
+      if re.match(regex, input_file_name) is not None:
+        return input_file_pattern
+    return None
+
+  def _input_file_pattern_to_regex(self, input_file_pattern: str) -> str:
+    """
+    Transforms a string of the form "a*b.c" to the regex "a.*b\.c".
```

Comment:
why need the transform here?

Reply:
Because we do regex matching internally and the literal "a*b.c" is not one. It is a simple glob-like wildcard pattern which won't work for regex matching.